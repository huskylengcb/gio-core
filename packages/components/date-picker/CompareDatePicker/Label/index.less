.compare-date-picker-label-wrapper {
  display: inline-block;
  height: 32px;
  line-height: 32px;
  transition: 0.2s;
  padding: 0 9px;
  background-color: #FAFAFA;
  border: 1px solid #F2F2F2;
  cursor: pointer;
  outline: none;
  transition: all 0.3s;
  font-size: 14px;
  &.focus {
    background-color: #3F4F9F;
    color: white;
  }
  &:focus, &:hover {
    border-color: #220968;
  }
}