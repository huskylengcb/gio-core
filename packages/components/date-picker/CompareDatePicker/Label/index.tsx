import React from 'react';
// import Button from 'giodesign/button';
import format from '../../Range/util/format';
import Gap from '@gio-design/components/lib/gap';
import Icon from '@gio-design/components/lib/icon';
import flattenDateRange from '../../flattenDate';
import './index.less'
interface  Props {
  label: string,
  onClick?: () => void,
  className?: string,
  focus: boolean,
  timeRange: string,
  showArrow?: boolean
}

const Label = (props: Props) => {
  const { startTime, endTime } = flattenDateRange(props.timeRange)
  const startStr = format(startTime)
  const endStr = format(endTime)
  return (
    <div
      className={`compare-date-picker-label-wrapper ${props.className} ${props.focus ? 'focus' : ''}`}
      onClick={props.onClick}
    >
      {props.label && (<span className='text' title={props.label}>{props.label}</span>)}
      <Gap width={5} />
      {props.timeRange ? startStr === endStr ? startStr : `${startStr} - ${endStr}` : null}
      {props.showArrow ? <Icon className={`icon arrow_down ${props.focus ? 'focus' : ''}`} name='gicon-arrow-down' fill={props.focus ? '#fff' : '#666'} /> : undefined}
    </div>
  )
};

export default Label
