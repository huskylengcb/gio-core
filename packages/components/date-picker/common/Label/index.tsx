import React from 'react';
import Icon from '@gio-design/icon';
import './index.less';

interface  Props {
    label: string,
    onClick: () => void,
    className?: string,
    focus: boolean,
    placeholder?: string
}

const Label = (props: Props) => (
    <button
        className={`gio-datepicker-label ${props.className || ''} ${props.focus ? 'focus' : ''}`}
        onClick={props.onClick}
        type='button'
    >
        <Icon className='icon calendar' type='calendar' fill='#7F7583' />
        {props.label && (<span className='text' title={props.label}>{props.label}</span>)}
        <Icon className={`icon arrow_down ${props.focus ? 'focus' : ''}`} type='down' fill='#666' />
    </button>
);

export default Label
