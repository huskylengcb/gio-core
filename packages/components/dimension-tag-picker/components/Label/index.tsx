import React from 'react';
import Tooltip from 'antd/lib/tooltip';
import Icon from 'giodesign/icon';

const iconStyle = { fill: '#F63434' };

const InvalidTip = (
  <Tooltip title='与已选择的某些事件或指标组合后无意义'>
    <span>
      <Icon name='gicon-explain' svgStyle={iconStyle} />
    </span>
  </Tooltip>
);

const Label = ({
  children,
  isInvalid
}) => isInvalid ? (<span>无效维度 {InvalidTip}</span>) : children;

export default Label;
