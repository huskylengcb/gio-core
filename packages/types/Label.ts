import Base from './Base';

export default interface Label extends Base {
  color: 'red' | 'pink' | 'purple' | 'deep-purple'
  | 'deep-orange' | 'orange' | 'amber' | 'yellow' | 'lime'
  | 'light-green' | 'green' | 'teal' | 'cyan' | 'light-blue'
  | 'blue' | 'indigo' | 'blue-grey' | 'brown' | 'grey' | 'deep-grey'
}
