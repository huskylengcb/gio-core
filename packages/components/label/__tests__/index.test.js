import React from 'react';
import Label from '../';
import renderer from 'react-test-renderer';

test('label', () => {
  const label = renderer.create(
    <Label>hello</Label>, {
      createNodeMock: (element) => ({
        scrollWidth: 0
      })
    }
  );
  let tree = label.toJSON();
  expect(tree).toMatchSnapshot();
});