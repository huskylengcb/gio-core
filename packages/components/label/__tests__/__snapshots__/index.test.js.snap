// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`label 1`] = `
<span
  className="gio-label light nowrap"
>
  <span
    className="gio-label-inner"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    hello
  </span>
</span>
`;
